components/brave_wallet_ui/common/slices/endpoints/on-ramp.endpoints.ts

```diff
@@ -0,0 +1,206 @@
+          const {
+            tokens: rampTokenOptions,
+            nativeAssets: rampNativeAssetOptions
+          } = getNativeTokensFromList(rampAssetOptions)
+
+          const {
+            tokens: sardineTokenOptions,
+            nativeAssets: sardineNativeAssetOptions
+          } = getNativeTokensFromList(sardineAssetOptions)
+
+          const {
+            tokens: transakTokenOptions,
+            nativeAssets: transakNativeAssetOptions
+          } = getNativeTokensFromList(transakAssetOptions)
+
+          const {
+            tokens: stripeTokenOptions,
+            nativeAssets: stripeNativeAssetOptions
+          } = getNativeTokensFromList(stripeAssetOptions)
+
+          // separate BAT from other tokens
+          const {
+            bat: rampBatTokens,
+            nonBat: rampNonBatTokens
+          } = getBatTokensFromList(rampTokenOptions)
+
+          const {
+            bat: sardineBatTokens,
+            nonBat: sardineNonBatTokens
+          } = getBatTokensFromList(sardineTokenOptions)
+
+          const {
+            bat: transakBatTokens,
+            nonBat: transakNonBatTokens
+          } = getBatTokensFromList(transakTokenOptions)
+
+          const {
+            bat: stripeBatTokens,
+            nonBat: stripeNonBatTokens
+          } = getBatTokensFromList(stripeTokenOptions)
+
+          // sort lists
+          // Move Gas coins and BAT to front of list
+          const sortedRampOptions =
+            [
+              ...rampNativeAssetOptions,
+              ...rampBatTokens,
+              ...rampNonBatTokens
+            ]
+          const sortedSardineOptions =
+            [
+              ...sardineNativeAssetOptions,
+              ...sardineBatTokens,
+              ...sardineNonBatTokens
+            ]
+          const sortedTransakOptions =
+            [
+              ...transakNativeAssetOptions,
+              ...transakBatTokens,
+              ...transakNonBatTokens
+            ]
+          const sortedStripeOptions =
+            [
+              ...stripeNativeAssetOptions,
+              ...stripeBatTokens,
+              ...stripeNonBatTokens
+            ]
+
+          const results = {
+            rampAssetOptions: sortedRampOptions,
+            sardineAssetOptions: sortedSardineOptions,
+            transakAssetOptions: sortedTransakOptions,
+            stripeAssetOptions: sortedStripeOptions,
+            allAssetOptions: getUniqueAssets([
+              ...sortedRampOptions,
+              ...sortedSardineOptions,
+              ...sortedTransakOptions,
+              ...sortedStripeOptions
+            ])
+          }
+
+          return {
+            data: results
+          }
+        } catch (error) {
+          const errorMessage = `Unable to fetch onRamp assets: ${error}`
+          console.log(errorMessage)
+          return {
+            error: errorMessage
+          }
+        }
+      }, providesTags: (results, error, _arg) => {
```

Comment:
Results is not used here
```suggestion
      }, providesTags: (_results, error, _arg) => {
```